src/lib/offline-api/validator/schema/schema-validation.ts

```diff
@@ -292,6 +293,61 @@ const validateFields = function (
       }
     }
 
+    // Field `allowedResources` errors
+    if (path[1] === 'allowedResources') {
```

Comment:
Is this the same as above? If so why not use the same condition for both?
```
if (prop.startsWith('allowedResources'))
```

Reply:
Either works. I can change it if you prefer this one.

Reply:
I understand. I don't have a preference but I thought maybe the author of the snippet above did it for some reason. 

Let me explain my point. When you're reading the code from above you understand the first condition and then you come to the second condition which is cohesive to the first one but looks different. Then as a reader you have more to process. I think the reader would have less cognitive load if the conditions were written in the same style. 

If there is no reason for `const prop = fieldNames.join('.')` and `prop.startsWith('defaultValue')` personally I prefer `path[1]` because there is less to process.

Reply:
Yes, I also prefer using `path[1]` because it’s more precise but I changed it to `startsWith` now to avoid touching the other code.